src/main/java/com/commercetools/sync/products/utils/ProductVariantUpdateActionUtils.java

```diff
@@ -117,39 +119,114 @@
+                    });
+
+            filterCollection(newProductVariantImages, newVariantImage ->
+                    !oldImages.contains(newVariantImage))
+                    .forEach(newImage -> {
+                        updateActions.add(AddExternalImage.ofVariantId(oldProductVariantId, newImage, true));
+                        updatedOldImages.add(newImage);
+                    });
+            updateActions.addAll(buildMoveImageToPositionUpdateActions(oldProductVariantId,
+                    updatedOldImages, newImages));
+        }
+        return updateActions;
+    }
 
-        final Integer oldProductVariantId = oldProductVariant.getId();
+    /**
+     * Compares an old {@link List} of {@link Image}s and a new one and returns a {@link List} of
+     * {@link MoveImageToPosition}&lt;{@link Product}&gt; with the given {@code variantId}. If both the lists are
+     * identical, then no update action is needed and hence an empty {@link List} is returned.
+     *
+     * <p>This method expects the two lists two contain the same images only in different order. Therefore, be cautios
+     * that supplying lists of different (missing/extra) images could results in an index out of bounds exception on the
+     * new position of an image.
+     *
+     * <p><b>Note</b>: the solution is still not optimized and may contain {@link MoveImageToPosition} actions
+     * for items which are already on desired positions (after previous moves in the sequence). This will be
+     * re-optimized in the next releases.
+     *
+     * @param variantId the variantId for the {@link MoveImageToPosition} update actions.
+     * @param oldImages the old list of images.
+     * @param newImages the new list of images.
+     * @return a list that contains all the update actions needed, otherwise an empty list if no update actions are
+     *         needed.
+     * @throws IllegalArgumentException if arrays have different size or different items.
+     */
+    public static List<MoveImageToPosition> buildMoveImageToPositionUpdateActions(
+            final int variantId,
+            @Nonnull final List<Image> oldImages,
+            @Nonnull final List<Image> newImages) throws IllegalArgumentException {
+
+        if (oldImages.size() != newImages.size()) {
+            throw new IllegalArgumentException(
+                format("Old and new image lists must have the same size, but they have %d and %d respectively",
+                    oldImages.size(), newImages.size()));
+        }
+
+        final int SIZE = oldImages.size();
+
+        // optimization: to avoid multiple linear image index searching in the loop below - create an [image -> index]
+        // map. This avoids quadratic order of growth of the implementation for large arrays.
+        final Map<Image, Integer> imageIndexMap = new HashMap<>(SIZE);
+        int index = 0;
+        for (Image newImage : newImages) {
+            imageIndexMap.put(newImage, index++);
+        }
```

Comment:
i added some optimization: created new image indexes cache to have constant access in line:
```java
final Integer newIndex = ofNullable(imageIndexMap.get(oldImage))
```

where before it was linear operation (inside loop)